test/unit/spec/common/cellComponents/tabs/jobStateList-Spec.js

```diff
@@ -65,357 +188,473 @@ define([
                     });
+                    expect($nextRow.find('.kb-log__logs_title')[0].textContent).toContain('Logs');
+
+                    // click on another row
+                    $currentRow = $(rows[5]);
+                    await TestUtil.waitForElementState(
+                        container.querySelector('tbody'),
+                        () => {
+                            return (
+                                container.querySelectorAll('.vertical_collapse--open').length === 2
+                            );
+                        },
+                        () => {
+                            $currentRow.click();
+                        }
+                    );
+                    expect(container.querySelectorAll('tbody tr').length).toEqual(rows.length + 2);
+
+                    // the current row should be selected and have the class 'vertical_collapse--open'
+                    expect($currentRow[0]).toHaveClass('vertical_collapse--open');
+                    expect($currentRow[0]).toHaveClass(`${cssBaseClass}__row--selected`);
+
+                    // the next row is a log viewer row
+                    $nextRow = $currentRow.next();
+                    ['odd', 'even'].forEach((cls) => {
+                        expect($nextRow[0]).not.toHaveClass(cls);
+                    });
+                    expect($nextRow.find('.kb-log__logs_title')[0].textContent).toContain('Logs');
+
+                    // click again to remove the row
+                    await TestUtil.waitForElementState(
+                        container.querySelector('tbody'),
+                        () => {
+                            return (
+                                container.querySelectorAll('.vertical_collapse--open').length === 1
+                            );
+                        },
+                        () => {
+                            $currentRow.click();
+                        }
+                    );
+                    expect(container.querySelectorAll('tbody tr').length).toEqual(rows.length + 1);
+
+                    // the current row is still selected but does not have the class 'vertical_collapse--open'
+                    expect($currentRow[0]).not.toHaveClass('vertical_collapse--open');
+                    expect($currentRow[0]).toHaveClass(`${cssBaseClass}__row--selected`);
+                    // the log viewer row has been removed, so the next row is a standard table row
+                    if ($currentRow[0].classList.contains('even')) {
+                        expect($currentRow.next()[0]).toHaveClass('odd');
+                    } else {
+                        expect($currentRow.next()[0]).toHaveClass('even');
+                    }
                 });
+            });
 
-                afterEach(async function () {
-                    await this.jobStateListInstance.stop();
+            describe('table buttons', () => {
```

Comment:
updated, based on the new action listeners that have been added to the table and the change in the way that the function to view results is implemented